task/import.go

```diff
@@ -0,0 +1,96 @@
+package task
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/golang/glog"
+	"golang.org/x/sync/errgroup"
+	ffprobe "gopkg.in/vansante/go-ffprobe.v2"
+)
+
+const (
+	fileUploadTimeout = 5 * time.Minute
+	// TODO: decide file name
+	videoFileName    = "video.mp4"
+	metadataFileName = "video.json"
+)
+
+func TaskImport(tctx *TaskContext) (interface{}, error) {
+	var (
+		ctx    = tctx.Context
+		srcURL = tctx.Params.Import.URL
+		osSess = tctx.osSession
+	)
+
+	req, err := http.NewRequestWithContext(ctx, "GET", srcURL, nil)
+	if err != nil {
+		glog.Errorf("Error creating http request url=%s err=%q", srcURL, err)
+		return nil, UnretriableError{err}
+	}
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("error on import request: %w", err)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode >= 300 {
+		err := fmt.Errorf("bad status code from import request: %d %s", resp.StatusCode, resp.Status)
+		if resp.StatusCode < 500 {
+			err = UnretriableError{err}
+		}
+		return nil, err
+	}
+
+	secondaryReader, pipe := io.Pipe()
```

Comment:
Need a third reader to calculate hash of the file (assuming ffprobe doesn't which I think it doesn't). File size should be already available on probe tho.

Also the hasher doesn't need to be another pipe. Could be just a reader wrapping the original body which accumulates the hash internally and allows querying afterwards.

Reply:
This makes me think about the hash of the file in the upload flow, with the import is easy as we got the full file, but on the upload we are proxying directly to the storage. On S3 we got the `Etag` which is infact an md5, so there should be in the runner or the API a service to retrieve it, wdyt? In any case, a sha256sum would be probably harder to get without having the full file?

Reply:
Btw, apparently it's not guaranteed that the `Etag` on the S3 API is an MD5. In the case of Filebase, sometimes it will be just an object UUID (when the file is saved with a multipart upload, which is how we are doing now).